juju/juju
# [JUJU-2972] Maintain secret permissions properly

When removing an application or unit, we were not also deleting any corresponding secret permission record. We were deleting an records where the entity matched the permission scope, but not the subject.

This fixes that plus adds an upgrade step to clean up any orphaned records.

## QA steps

create and share a secret
from the consumer, read the secret
delete the consumer
use mongo shell to check the secret permission record has been removed from "secretPermissions" collection
